### PR TITLE
feat: Snap Profile Image

### LINKED ISSUE #601: Feature Request: Take Profile Picture
**Summary:**
Taking profile picture while setting it is one of the standard features of android mobile application. 

Screenshots of other apps having similar feature:
Screenshot_20191213-135528_Android System

### PR DESCRIPTION
**Pull request by ReCodee**
## Issue Fix
Fixes #601 

## Description
Added an `Intent` for the Camera.

##

- [x]  Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by ReCodee** [comment]
@devansh-299 Please Review this PR.

**Comment by devansh-299** [comment]
@ReCodee the bottom sheet menu can possibly be modified to:
* Click profile picture
* Pick profile picture from device/gallery
* Remove profile picture
to give more clarity to the users about the corresponding actions

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java [hunk 1]
```diff
@@ -377,6 +388,24 @@ private void handleCropResult(@NonNull Intent result) {
         }
     }
 
+    public void snapProfilePicFromCamera() {
+
+        if (Build.VERSION.SDK_INT >= 23 && ContextCompat.checkSelfPermission(this,
+                Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+
+            // Permission is not granted
+            requestPermissions(new String[]{Manifest.permission.CAMERA}, REQUEST_CAMERA);
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java:413]: @ReCodee Permission handling is incomplete. Please specify the actions to be taken on the result of the user's request for the permission inside `onRequestPermissionsResult`.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java [hunk 2]
```diff
@@ -377,6 +388,24 @@ private void handleCropResult(@NonNull Intent result) {
...
+        } else {
+
+            // Permission has already been granted
+            Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+            startActivityForResult(intent, REQUEST_CAMERA);
+        }
+
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java]: Un-necessary empty lines

**Review by devansh-299 — CHANGES_REQUESTED** [review]
Also can you attach a screen recording of your changes to help others judge your changes easily?

**Comment by ReCodee** [comment]
> @ReCodee the bottom sheet menu can possibly be modified to:
> 
> * Click profile picture
> * Pick profile picture from device/gallery
> * Remove profile picture
>   to give more clarity to the users about the corresponding actions

Thank you, I've updated the PR.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java (hunk 1 above)
Comment (ReCodee) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java:413]: It completely slipped from my head, thank you for pointing this out. I've made the changes.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java [hunk 3]
```diff
@@ -268,6 +270,23 @@ public void hideKeyboard() {
...
+                    // permission was granted, yay! Do the
+                    // camera-related task you need to do.
+                    Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                    startActivityForResult(intent, REQUEST_CAMERA);
+
+                } else {
+                    // permission denied, boo! Disable the
+                    // functionality that depends on this permission.
+                    Toaster.showToast(this,
+                            Constants.NEED_CAMERA_PERMISSION_TO_CLICK_PROFILE_PICTURE);
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java]: @ReCodee For any user-visible message/text we should use a string resource object as we can easily provide a multi-language alternative for them when needed. For any internal workings of the application we can use `Constants` as the above use case doesn't apply to them.

I just saw this hasn't been followed by the _need storage message_ text also and possibly at many other places too. A separate issue can address this, but yeah we can avoid this for any new additions

File: mifospay/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -175,8 +175,9 @@
     <string name="pin_code">Pin Code</string>
     <string name="state">State</string>
     <string name="no_transactions_found">No transactions found</string>
-    <string name="change_profile_picture">Change Profile Picture</string>
-    <string name="remove_profile_picture">Remove Profile Picture</string>
+    <string name="click_profile_picture">Click profile picture</string>
+    <string name="change_profile_picture">Pick profile picture from device/gallery</string>
```
Comment (devansh-299) [inline mifospay/src/main/res/values/strings.xml]: Let's change the message to `Pick profile picture from device`. 
Earlier I was trying to say that we can use any one of them (device or gallery) for the text that will be displayed to the user

**Review by devansh-299 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java (hunk 3 above)
Comment (ReCodee) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java]: Thank you, I'll keep this in mind from now on. Btw can I make an issue to resolve this matter?

File: mifospay/src/main/res/values/strings.xml (hunk 4 above)
Comment (ReCodee) [reply mifospay/src/main/res/values/strings.xml]: Thanks, I've made the changes.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java (hunk 3 above)
Comment (devansh-299) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java]: Yeah, if this problem persists at many places, a pull request can be addressed to solving this

**Review by devansh-299 — APPROVED** [review]
LGTM, nice work @ReCodee

**Comment by devansh-299** [comment]
There's a merge conflict with this pull request, please resolve it

**Comment by ReCodee** [comment]
> There's a merge conflict with this pull request, please resolve it

Done, Thanks for the review :)

**Comment by s-ayush2903** [comment]
@ReCodee Can you please update(synchronize) your branch with the target one, doing so will trigger the CI checks and improve the quality of same also.

**Comment by ReCodee** [comment]
> @ReCodee Can you please update(synchronize) your branch with the target one, doing so will trigger the CI checks and improve the quality of same also.

Done!, Thanks for the help :).

**Review by s-ayush2903 — COMMENTED** [review]
It can be better if we fetch dimensions from from their corresponding resource file, rest looks cool to me.

**Review by s-ayush2903 — APPROVED** [review]
Good work, thanks for contribution!

**Pull request merged**